pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16282778?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16282778?buildTab=artifacts#/) on master @ [7afbca49760b34c2636971cdadf63e33aa23afdc](https://github.com/cockroachdb/cockroach/commits/7afbca49760b34c2636971cdadf63e33aa23afdc):


```
        sql.plan.type.force-generic
    datadriven.go:144: 
        testdata/telemetry/generic:81:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:85:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:89:
        exec [0 args]
        SET plan_cache_mode = auto
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:94:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.auto-generic
    datadriven.go:144: 
        testdata/telemetry/generic:101:
        feature-usage [0 args]
        EXECUTE p_join(2)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:106:
        feature-usage [0 args]
        EXECUTE p_join(3)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:111:
        feature-usage [0 args]
        EXECUTE p_join(4)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:116:
         EXECUTE p_join(5)
        expected:
        sql.plan.type.auto-custom
        
        found:
        sql.plan.type.auto-custom
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (0.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40769

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://tanzanite.cluster.engflow.com/invocations/default/add44392-9801-4b21-9a2e-97011b798a39?testReportRun=6&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL3RlbGVtZXRyeWNjbDp0ZWxlbWV0cnljY2xfdGVzdA==) on master @ [7a1bf52daa3f3dcaa434fd66d2b3c960a3c9c2a5](https://github.com/cockroachdb/cockroach/commits/7a1bf52daa3f3dcaa434fd66d2b3c960a3c9c2a5):


```
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:89:
        exec [0 args]
        SET plan_cache_mode = auto
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:94:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.auto-generic
    datadriven.go:144: 
        testdata/telemetry/generic:101:
        feature-usage [0 args]
        EXECUTE p_join(2)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:106:
        feature-usage [0 args]
        EXECUTE p_join(3)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:111:
        feature-usage [0 args]
        EXECUTE p_join(4)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:116:
        feature-usage [0 args]
        EXECUTE p_join(5)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:122:
        feature-usage [0 args]
        EXECUTE p_join(6)
        ----
        sql.plan.type.auto-generic
    datadriven.go:144: 
        testdata/telemetry/generic:128:
         SELECT * FROM kv WHERE v = 100
        expected:
        
        found:
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (1.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://tanzanite.cluster.engflow.com/invocations/default/74576e32-820b-47a6-841f-ac24652f0abb?testReportRun=13&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL3RlbGVtZXRyeWNjbDp0ZWxlbWV0cnljY2xfdGVzdA==) on master @ [78fa126dfc749e620c28531c1659e35bf98034ec](https://github.com/cockroachdb/cockroach/commits/78fa126dfc749e620c28531c1659e35bf98034ec):


```
    datadriven.go:144: 
        testdata/telemetry/generic:66:
        exec [0 args]
        SET plan_cache_mode = force_generic_plan
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:70:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.force-generic
    datadriven.go:144: 
        testdata/telemetry/generic:75:
        feature-usage [0 args]
        EXECUTE p_join(0)
        ----
        sql.plan.type.force-generic
    datadriven.go:144: 
        testdata/telemetry/generic:81:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:85:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:89:
        exec [0 args]
        SET plan_cache_mode = auto
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:94:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.auto-generic
    datadriven.go:144: 
        testdata/telemetry/generic:101:
         EXECUTE p_join(2)
        expected:
        sql.plan.type.auto-custom
        
        found:
        sql.plan.type.auto-custom
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (0.44s)
=== RUN   TestTelemetry/generic
    --- FAIL: TestTelemetry/generic (6.38s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=13</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128171 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot T-observability branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16304408?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16304408?buildTab=artifacts#/) on master @ [6ae5e64c3ddc3a0e1bbb304f2930f0bbdc184d86](https://github.com/cockroachdb/cockroach/commits/6ae5e64c3ddc3a0e1bbb304f2930f0bbdc184d86):


```
        SELECT * FROM kv WHERE k = $1
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:24:
        exec [0 args]
        PREPARE p_join AS
        SELECT * FROM kv
        JOIN ab ON a = k
        WHERE v = $1
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:31:
        feature-list [0 args]
        sql.plan.type.*
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:35:
        exec [0 args]
        SET plan_cache_mode = force_custom_plan
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:39:
        exec [0 args]
        SET index_recommendations_enabled = false
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:43:
        exec [0 args]
        SET application_name = 'testuser'
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:47:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:52:
        feature-usage [0 args]
        EXECUTE p_join(1)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:58:
         SELECT * FROM kv WHERE v = 100
        expected:
        
        found:
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (0.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128171 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot T-observability branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://tanzanite.cluster.engflow.com/invocations/default/172b1fb5-b45b-40d5-9b03-da50188c2b2a?testReportRun=14&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL3RlbGVtZXRyeWNjbDp0ZWxlbWV0cnljY2xfdGVzdA==) on master @ [356e1cc24a67d7e9a1e637b98d1664794f70fa03](https://github.com/cockroachdb/cockroach/commits/356e1cc24a67d7e9a1e637b98d1664794f70fa03):


```
    datadriven.go:144: 
        testdata/telemetry/generic:85:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:89:
        exec [0 args]
        SET plan_cache_mode = auto
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:94:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.auto-generic
    datadriven.go:144: 
        testdata/telemetry/generic:101:
        feature-usage [0 args]
        EXECUTE p_join(2)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:106:
        feature-usage [0 args]
        EXECUTE p_join(3)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:111:
        feature-usage [0 args]
        EXECUTE p_join(4)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:116:
        feature-usage [0 args]
        EXECUTE p_join(5)
        ----
        sql.plan.type.auto-custom
    datadriven.go:144: 
        testdata/telemetry/generic:122:
         EXECUTE p_join(6)
        expected:
        sql.plan.type.auto-generic
        
        found:
        sql.plan.type.auto-generic
        sql.plan.type.force-custom
        --- FAIL: TestTelemetry/generic/server (0.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128171 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot T-observability branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/telemetryccl/telemetryccl_test.TestTelemetry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16356305?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16356305?buildTab=artifacts#/) on master @ [7917e91eadb2f622faadf635a3e13c384e42070e](https://github.com/cockroachdb/cockroach/commits/7917e91eadb2f622faadf635a3e13c384e42070e):


```
        exec [0 args]
        SET plan_cache_mode = force_custom_plan
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:39:
        exec [0 args]
        SET index_recommendations_enabled = false
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:43:
        exec [0 args]
        SET application_name = 'testuser'
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:47:
        feature-usage [0 args]
        EXECUTE p_pk(100)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:52:
        feature-usage [0 args]
        EXECUTE p_join(1)
        ----
        sql.plan.type.force-custom
    datadriven.go:144: 
        testdata/telemetry/generic:58:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:62:
        feature-usage [0 args]
        SELECT * FROM kv WHERE v = 100
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:66:
        exec [0 args]
        SET plan_cache_mode = force_generic_plan
        ----
    datadriven.go:144: 
        testdata/telemetry/generic:70:
         EXECUTE p_pk(100)
        expected:
        sql.plan.type.force-generic
        
        found:
        sql.plan.type.force-custom
        sql.plan.type.force-generic
        --- FAIL: TestTelemetry/generic/server (0.15s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128171 pkg/ccl/telemetryccl/telemetryccl_test: TestTelemetry failed [C-test-failure O-robot P-1 T-observability branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTelemetry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/telemetryccl/testdata/telemetry/generic
+++ b/pkg/ccl/telemetryccl/testdata/telemetry/generic
@@ -29,7 +29,7 @@ WHERE v = $1
 ----
 
 feature-list
-sql.plan.type.*
+sql.plan.type.force-custom
 ----
 
 exec
@@ -63,6 +63,10 @@ feature-usage
 SELECT * FROM kv WHERE v = 100
 ----
 
+feature-list
+sql.plan.type.force-generic
+----
+
 exec
 SET plan_cache_mode = force_generic_plan
 ----
@@ -90,11 +94,19 @@ exec
 SET plan_cache_mode = auto
 ----
 
+feature-list
+sql.plan.type.auto-generic
+----
+
 # If the placeholder fast-path is used, the plan is always generic.
 feature-usage
 EXECUTE p_pk(100)
 ----
 sql.plan.type.auto-generic
+
+feature-list
+sql.plan.type.auto-custom
+----
 
 # The first five executions of p_join have custom plans while establishing an
 # average cost. One of the custom executions occurred above.
@@ -117,6 +129,11 @@ feature-usage
 EXECUTE p_join(5)
 ----
 sql.plan.type.auto-custom
+
+
+feature-list
+sql.plan.type.auto-generic
+----
 
 # The sixth execution uses a generic plan.
 feature-usage

